elastic/beats
# add_cloud_metadata - Retrieve AKS cluster name and id

## Proposed commit message

- WHAT: Enhance `add_cloud_metadata` processor with `orchestrator.cluster.name` and `orchestrator.cluster.id` for AKS clusters.
- WHY:  It allows filtering of events based on the AKS cluster the vm belongs to.

## Checklist





## How to test this PR locally



- Create an AKS cluster
- Run Metricbeat or Filebeat with [sleep](https://github.com/elastic/beats/blob/90228e66040f055f8bcd5395e7161a3d3f26018e/metricbeat/module/kubernetes/_meta/test/docs/01_playground/metricbeat.yaml#L155) command and correct values of following additional env vars
 ```
- name: TENANT_ID
  value: ""
- name: CLIENT_ID
  value: ""
- name: CLIENT_SECRET
  value: ""
 ```
In order to acquire a CLIENT_SECRET and CLIENT_ID follow the [app_registration](https://docs.elastic.co/integrations/azure_billing#app-registration) instructions.
- Build this pr's version of metricbeat/filebeat with `GOOS=linux GOARCH=amd64 go build`
- Copy the metricbeat/filebeat binary in metricbeat/filebeat pod with `kubectl cp ./metricbeat metricbeat-9lkhf:/usr/share/metricbeat/ -n kube-system`
- Exec in the pod and run `./metricbeat  -c /etc/metricbeat.yml -e`
- Check in Kibana the fields `orchestrator.cluster.name` and `orchestrator.cluster.id` being populated

## Related issues

- Closes https://github.com/elastic/beats/issues/33081

## Use cases



## Screenshots



<img width="1776" alt="orchestrator" src="https://github.com/elastic/beats/assets/26270880/a5fb4f1b-011e-4942-812f-04c4f92380bc">


## Logs
